src/rubysys/value.rs

```diff
@@ -1,81 +1,65 @@
 use std::{convert::From, mem};
 
-use crate::rubysys::{
+use super::{
     constant,
     types::{InternalValue, RBasic},
 };
 
 const SPECIAL_SHIFT: usize = 8;
 
-#[cfg(target_pointer_width = "32")]
-pub enum RubySpecialConsts {
-    False = 0,
-    True = 0x02,
-    Nil = 0x04,
-    Undef = 0x06,
-}
+use rb_sys::{ruby_special_consts, ruby_value_type};
 
-#[cfg(target_pointer_width = "32")]
-pub enum RubySpecialFlags {
-    ImmediateMask = 0x03,
-    FixnumFlag = 0x01,
-    FlonumMask = 0x00,
-    FlonumFlag = 0x02,
-    SymbolFlag = 0x0e,
-}
-
-#[cfg(target_pointer_width = "64")]
+// I assume pointer width is handled by rb_sys;
 pub enum RubySpecialConsts {
-    False = 0,
-    True = 0x14,
-    Nil = 0x08,
-    Undef = 0x34,
+    False = ruby_special_consts::RUBY_Qfalse as isize,
+    True = ruby_special_consts::RUBY_Qtrue as isize,
+    Nil = ruby_special_consts::RUBY_Qnil as isize,
+    Undef = ruby_special_consts::RUBY_Qundef as isize,
 }
 
-#[cfg(target_pointer_width = "64")]
 pub enum RubySpecialFlags {
-    ImmediateMask = 0x07,
-    FixnumFlag = 0x01,
-    FlonumMask = 0x03,
-    FlonumFlag = 0x02,
-    SymbolFlag = 0x0c,
+    ImmediateMask = ruby_special_consts::RUBY_IMMEDIATE_MASK as isize,
+    FixnumFlag = ruby_special_consts::RUBY_FIXNUM_FLAG as isize,
+    FlonumMask = ruby_special_consts::RUBY_FLONUM_MASK as isize,
+    FlonumFlag = ruby_special_consts::RUBY_FLONUM_FLAG as isize,
+    SymbolFlag = ruby_special_consts::RUBY_SYMBOL_FLAG as isize,
 }
 
 // #[link_name = "ruby_value_type"]
 #[derive(Debug, PartialEq)]
-#[repr(C)]
+#[repr(u32)]
 pub enum ValueType {
-    None = 0x00,
-
-    Object = 0x01,
-    Class = 0x02,
-    Module = 0x03,
-    Float = 0x04,
-    RString = 0x05,
-    Regexp = 0x06,
-    Array = 0x07,
-    Hash = 0x08,
-    Struct = 0x09,
-    Bignum = 0x0a,
-    File = 0x0b,
-    Data = 0x0c,
-    Match = 0x0d,
-    Complex = 0x0e,
-    Rational = 0x0f,
-
-    Nil = 0x11,
-    True = 0x12,
-    False = 0x13,
-    Symbol = 0x14,
-    Fixnum = 0x15,
-    Undef = 0x16,
-
-    IMemo = 0x1a,
-    Node = 0x1b,
-    IClass = 0x1c,
-    Zombie = 0x1d,
-
-    Mask = 0x1f,
+    None = ruby_value_type::RUBY_T_NONE as u32,
```

Comment:
I think we can remove these casts (they shouldnt be neccesary?)

Reply:
I think they are since `ValueType` is part of the public API of the `rubysys` module. And it uses those names for the enum items. Please let me know if you see another way.